memory/test/test_memory.py

```diff
@@ -33,28 +33,40 @@ def setUp(self):
         token = uuid.uuid4()
         self.test_path = f'gs://{bucket_name}/memory-tests/{token}'
 
-        self.fs = GCS(concurrent.futures.ThreadPoolExecutor(), project=os.environ['PROJECT'])
+        self.fs = GoogleStorageAsyncFS(project=os.environ['PROJECT'])
+        self.sem = asyncio.Semaphore(50)
         self.client = BlockingMemoryClient(fs=self.fs)
         self.temp_files = set()
 
     def tearDown(self):
-        async_to_blocking(self.fs.delete_gs_files(self.test_path))
+        async def delete_files(url):
+            async with self.sem:
+                await self.fs.rmtree(self.sem, url)
+
+        async_to_blocking(delete_files(self.test_path))
```

Comment:
It seems simpler to use:

```
async_to_blocking(self.fs.rmtree(None, url)
```
It also saves unnecessarily adding `self.sem` as a class field, where it seems to be unused except for in the tearDown.